### PR TITLE
Adding recipe for DAGMC

### PR DESCRIPTION
**Pull request by ngranda**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by ngranda** [comment]
Can you please confirm that you are willing to be a maintainer in this repo by commenting in this PR.
@gonuke, @pshriwise,  @ljacobson64, @makeclean, @kkiesling, @bam241

**Comment by kkiesling** [comment]
I can confirm I am willing to be a maintainer

**Comment by ljacobson64** [comment]
I am willing to maintain

**Comment by pshriwise** [comment]
In.

**Comment by gonuke** [comment]
I'm happy to be a maintainer

File: recipes/dagmc/meta.yaml [hunk 1]
```diff
@@ -29,6 +29,8 @@ requirements:
     - python
     - hdf5
     - moab 5.*
+    - libblas
```
Comment (gonuke) [inline recipes/dagmc/meta.yaml]: What do we need BLAS for?

Comment (ngranda) [reply recipes/dagmc/meta.yaml]: I think that was the wrong one. I need libopenblas

**Comment by ngranda** [comment]
@conda-forge-admin, please rerender

**Comment by npavlovikj** [comment]
> @conda-forge-admin, please rerender

FYI, that command only works for PRs in feedstocks, https://conda-forge.readthedocs.io/en/latest/webservice.html#conda-forge-admin-please-rerender. You should be able to re-run the checks by clicking the button in the CI or adding a new commit.

**Comment by gonuke** [comment]
Thanks @npavlovikj - how is it possible to automatically generate more of the infrastructure, particularly the scripts to build locally?  (is it possible?)

**Comment by npavlovikj** [comment]
> Thanks @npavlovikj - how is it possible to automatically generate more of the infrastructure, particularly the scripts to build locally? (is it possible?)

You mean to test it locally? This might help, https://conda-forge.readthedocs.io/en/latest/testing.html#run-docker-tests-locally-for-staged-recipes.

**Comment by gonuke** [comment]
Thanks @npavlovikj - of course... I was too focused on the changed files, but the infrastructure I'm looking for is already in the master branch.

**Comment by bam241** [comment]
I am willing to maintain ! thx for taking care of this @ngranda !

**Comment by ngranda** [comment]
@conda-forge/staged-recipes, all the checks have passed. Can you let me know if there is anything else I need to do to move this forward?

**Comment by bam241** [comment]
@conda-forge/staged-recipes , all the checks have passed. 
Can you let us know if there is anything else I need to do to move this forward?

File: recipes/dagmc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - libopenblas
+    - zlib
+
+test:
+   commands:
+     - make_watertight --help
+
+about:
+  home: https://svalinn.github.io/DAGMC/
+  license: BSD
```
Comment (xhochy) [inline recipes/dagmc/meta.yaml]: Which kind of BSD license? Please use the SPDX classifier here and add `license_family: BSD`

Comment (xhochy) [inline recipes/dagmc/meta.yaml]: Which kind of BSD license? Please use the SPDX classifier here and add `license_family: BSD`

File: recipes/dagmc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{% set name = "dagmc" %}
+{% set version = "3.2" %}
```
Comment (ljacobson64) [inline recipes/dagmc/meta.yaml]: I'd like to propose that the version be 3.2.0 instead of just 3.2

File: recipes/dagmc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - {{ compiler("fortran") }}
+  host:
+    - python
+    - hdf5
+    - moab 5.*
+  run:
+    - python
+    - hdf5
+    - moab 5.*
+    - libopenblas
```
Comment (scopatz) [inline recipes/dagmc/meta.yaml]: Please follow these instructions for linking to BLAS: https://conda-forge.org/docs/maintainer/knowledge_base.html#blas

**Comment by scopatz** [comment]
Very cool to see this coming into conda-forge

File: recipes/dagmc/meta.yaml (hunk 4 above)
Comment (ljacobson64) [reply recipes/dagmc/meta.yaml]: Can you clarify? `libblas` doesn't work here

Comment (npavlovikj) [reply recipes/dagmc/meta.yaml]: I think the suggestion is to add `libblas` in the `host` section, and remove `libopenblas` from the `run` section.

**Comment by scopatz** [comment]
Is this ready for review?

**Comment by gonuke** [comment]
yes please!

File: recipes/dagmc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/svalinn/DAGMC/archive/v{{ version }}-rc.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+  detect_binary_files_with_prefix: true
```
Comment (scopatz) [inline recipes/dagmc/meta.yaml]: Is this needed?

File: recipes/dagmc/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+extra:
+  recipe-maintainers:
+    - gonuke
+    - pshriwise
+    - ljacobson64
+    - makeclean
+    - kkiesling
+    - bam241
+    - ngranda
```
Comment (scopatz) [inline recipes/dagmc/meta.yaml]: @gonuke @pshriwise @ljacobson64 @makeclean @kkiesling @bam241 are you OK being maintainers?

Comment (gonuke) [reply recipes/dagmc/meta.yaml]: Everyone other than @makeclean has indicated their willingness in comments above.

Comment (gonuke) [reply recipes/dagmc/meta.yaml]: Everyone other than @makeclean has indicated their willingness in comments above.

**Comment by makeclean** [comment]
I'd love to say yes, but my bandwidth is negative right now
________________________________

**Comment by ngranda** [comment]
@makeclean, no worries. Thanks for letting us know

**Comment by scopatz** [comment]
Awesome thanks All!

**Pull request merged**